openmeetings-core/src/main/java/org/apache/openmeetings/core/remote/KStream.java

```diff
@@ -162,15 +162,39 @@ public void onError(Throwable cause) throws Exception {
 		});
 	}
 
+	/**
+	 * Invoked in case stream stops to decide on if this stream is worth stopping.
+	 *
+	 * Stop broadcast in case:
+	 *  - Audio only stream and audio has been detected as not streaming anymore
+	 *  - Video or Data
+	 *
+	 * @param mediaType the MediaType that stopped flowing
+	 * @return true in case this stream should be dropped
+	 */
+	private boolean checkFlowOutEventForStopping(MediaType mediaType) {
+		if (MediaType.AUDIO == mediaType) {
```

Comment:
This looks extremely cumbersome :(

Could you please change this method to be: `return MediaType.AUDIO != mediaType || !hasVideo;` ?

Reply:
I can, I just had to think for almost 15min around if those 3 use-cases:

1. It should return false (not stopBroadCast) in case MediaType.AUDIO == mediaType and hasVideo == true
2. It should return true (do stopBroadCast)in case MediaType.AUDIO == mediaType and hasVideo == false
3. It should return true (do stopBroadCast) in case MediaType.AUDIO != mediaType

are correct with this statement. I think this double negation with ||, I don't think a lot of people find it easy to translate it actual behaviour.

Reply:
Done

Reply:
If you find this condition hard to read you can rewrite the method as 
```
protected boolean checkFlowOutEventForRunning(MediaType mediaType) {
		return MediaType.AUDIO == mediaType && hasVideo;
}
```

and add negation while using:
`if (MediaFlowState.NOT_FLOWING == evt.getState() && !checkFlowOutEventForRunning(evt.getMediaType()))`

is it looks better?